core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2603,4 +2603,14 @@ package object config {
       .stringConf
       .toSequence
       .createWithDefault("org.apache.spark.sql.connect.client" :: Nil)
+
+  private[spark] val LEGACY_ABORT_STAGE_AFTER_CANCEL_TASKS =
+    ConfigBuilder("spark.scheduler.stage.legacyAbortAfterCancelTasks")
+      .doc("Whether to abort a stage after TaskScheduler.cancelTasks(). This is used to restore " +
+        "the original behavior in case there are any regressions after abort stage is removed " +
+        "from TaskScheduler.cancelTasks()")
+      .version("4.0.0")
+      .internal()
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
hmm, isn't 4.0 a good chance to enable this change? I don't think 4.1 will be a better chance than 4.0.

Reply:
Given this is a new change, might be good to be conservative about it.
We can change default in 4.1 or later, it does not need to be at major version.